src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs

```diff
@@ -299,8 +303,46 @@ public IActionResult FileUploadStart(SelectAddDraftApprenticeshipJourneyRequest
         [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
         public async Task<IActionResult> FileUploadStart(FileUploadStartViewModel viewModel)
         {
-            var request = await _modelMapper.Map<FileUploadReviewRequest>(viewModel);
-            return RedirectToAction(nameof(FileUploadReview), request);
+            var hasErrors = await ValidateBulkUploadData(viewModel.ProviderId, viewModel.Attachment);
+            if (hasErrors)
+            {
+               return RedirectToAction(nameof(FileUploadValidationErrors), new FileUploadValidateErrorRequest { ProviderId = viewModel.ProviderId });
+            }
+            else
+            {
+                var request = await _modelMapper.Map<FileUploadReviewRequest>(viewModel);
+                return RedirectToAction(nameof(FileUploadReview), request);
+            }
+        }
+
+        private async Task<bool> ValidateBulkUploadData(long providerId, IFormFile attachement)
+        {
+            var bulkValidate = new FileUploadValidateDataRequest { Attachement = attachement, ProviderId = providerId };
+            var response = await _mediator.Send(bulkValidate);
+            if (response.BulkUploadValidationErrors != null && response.BulkUploadValidationErrors.Count > 0)
+            {
+                TempData.Put("bulk-upload-errors", response);
+                return true;
+            }
+
+            return false;
+        }
+
+        [HttpGet]
+        [Route("add/file-upload/validate")]
+        [DasAuthorize(ProviderFeature.BulkUploadV2)]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        public async Task<IActionResult> FileUploadValidationErrors(FileUploadValidateErrorRequest request)
+        {
+            var errors = TempData.Get<BulkUploadValidateApiResponse>("bulk-upload-errors");
```

Comment:
Do we need to use tempdata? Could we just return the errors and pass them in the FileUploadValidateErrorRequest object?